### ISSUE TITLE
Image Load Failures lead to mismatch between DB entries and Distribution Graph

### ISSUE THREAD

**Issue by rodrigoberriel** [NVIDIA/DIGITS#1473]
### tl;dr

The distribution data (used to plot the Distribution Graph) is incorrect when images fail to load.

---
In case of image load failures (I only tested on Classification Datasets), there will be a mismatch between the **DB entries** and the **Distribution Graph** shown in the dataset page.

This happens because **DB entries** use `task.entries_count` that comes from the [number of images written to the database](https://github.com/NVIDIA/DIGITS/blob/master/digits/dataset/tasks/create_db.py#L212), and it takes the number of images loaded into account, i.e. DB entries consider only images loaded. On the other hand, the **Distribution Graph** uses `task.distribution_data()` that comes from [an information](https://github.com/NVIDIA/DIGITS/blob/master/digits/dataset/tasks/create_db.py#L194) that is calculated from [parsing the text file](https://github.com/NVIDIA/DIGITS/blob/master/digits/tools/create_db.py#L506), i.e. the Distribution Graph doesn't take load failures into account.

I think two actions are required:
1. Fix the Distribution Graph to show the correct number of images loaded on each class
2. Show a warning with the number of failures after finishing the job

`LoadImageError` can be thrown for [many reasons](https://github.com/NVIDIA/DIGITS/blob/master/digits/utils/image.py#L40):
- LoadImageError: "{image_path}" not found
- LoadImageError: IOError: image file is truncated (20 bytes not processed)
- LoadImageError: IOError: cannot identify image file
- etc.

I think `not found` errors are unexpected for the user and they should be checked during the initial parse. Also, these errors should stop the job<sup>1</sup>. I've made a test and for a text file with 200k image paths, it increased the time spent on [`_fill_load_queue()`](https://github.com/NVIDIA/DIGITS/blob/master/digits/tools/create_db.py#L465) ~15 seconds. The job itself took ~2 hours. I think it's worth it.

The other ones require `try-except` on `Image.open(...)` which might be too expensive for very long text files, i.e. very large databases. Therefore, the current behavior + displaying a message at the end (warning the user) must be fine.

<sup>1</sup> <sub>I lost ~2 hours of a database creation job to discover that the script I used to generate the text files had a bug and ~20% of the paths were broken. Besides that, I looked at the Distribution Graph and everything looked as I was expecting :(</sub>

---
DIGITS: 5.1-dev (I had these problems when using DIGITS 4 too)
Caffe: 0.15.14
Caffe flavor: NVIDIA
